tests/Integration/Endpoints/RecommendationsProxyEndpointTest.php

```diff
@@ -0,0 +1,126 @@
+
+		$this->wp_rest_server_global_backup        = $GLOBALS['wp_rest_server'] ?? null;
+		$this->rest_api_init_recommendations_proxy = function () {
+			$endpoint = new Recommendations_API_Proxy( new Parsely() );
+			$endpoint->run();
+		};
+		add_action( 'rest_api_init', $this->rest_api_init_recommendations_proxy );
+	}
+
+	/**
+	 * Reset globals.
+	 */
+	public function tearDown(): void {
+		parent::tearDown();
+		remove_action( 'rest_api_init', $this->rest_api_init_recommendations_proxy );
+		$GLOBALS['parsely'] = $this->parsely_global_backup;
+		// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedVariableFound
+		$GLOBALS['wp_rest_server'] = $this->wp_rest_server_global_backup;
+		delete_option( 'parsely' );
+		remove_filter( 'wp_parsely_enable_recommendations_endpoint', '__return_true' );
+	}
+
+	/**
+	 * Confirm the the route is registered.
+	 *
+	 * @covers Recommendations_API_Proxy::register_rest_route
+	 */
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+		$this->assertArrayHasKey( '/wp-parsely/v1/recommendations', $routes );
+		$this->assertSame( 1, count( $routes['/wp-parsely/v1/recommendations'] ) );
+		$this->assertEquals( array( 'GET' => true ), $routes['/wp-parsely/v1/recommendations'][0]['methods'] );
+	}
+
+	/**
+	 * Confirm that calls to `GET /wp-parsely/v1/recommendations` get results in the expected format.
+	 *
+	 * @covers Recommendations_API_Proxy::get_items
+	 */
+	public function test_get_items() {
+		$request  = new WP_REST_Request( 'GET', '/wp-parsely/v1/recommendations' );
+		$response = null;
+		self::mock_remote_network_request(
+			function () use ( &$response, $request ) {
+				$response = rest_get_server()->dispatch( $request );
+			},
+			'{"data":[{"image_url":"https:\/\/example.com\/img.png","title":"something","url":"https:\/\/example.com"},{"image_url":"https:\/\/example.com\/img2.png","title":"something2","url":"https:\/\/example.com\/2"}]}'
+		);
+
+		$this->assertSame( 200, $response->get_status() );
+		$this->assertEquals(
+			(object) array(
+				'data' => array(
+					(object) array(
+						'image_url' => 'https://example.com/img.png',
+						'title'     => 'something',
+						'url'       => 'https://example.com',
+					),
+					(object) array(
+						'image_url' => 'https://example.com/img2.png',
+						'title'     => 'something2',
+						'url'       => 'https://example.com/2',
+					),
+				),
+			),
+			$response->get_data()
+		);
+	}
+
+	/**
+	 * Confirm that calls to `GET /wp-parsely/v1/recommendations` gets and error and makes no remote call when the apikey is not populated in site options.
+	 *
+	 * @covers Recommendations_API_Proxy::get_items
+	 */
+	public function test_get_items_fails_without_apikey_set() {
+		delete_option( 'parsely' );
+		$request    = new WP_REST_Request( 'GET', '/wp-parsely/v1/recommendations' );
+		$response   = null;
+		$dispatched = TestCase::mock_remote_network_request(
```

Comment:
This uses `TestCase`, but the same call above uses `self`. Any reason for the inconsistency?